### PR TITLE
PersistentFSM and PersistenceQuery improvements

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
- Improved PersistentFSM documentation
- Use snapshot-after in config for periodical snapshots in PersistentFSM (https://github.com/akka/akka/pull/22605)
- Made `StateChangeEvent` and `PersistentFSMSnapshot` public.
- Changed behaviour of Recovery.none (https://github.com/akka/akka/pull/23043)
- Naming consistency AllPersistenceIds ->PersistenceIds (https://github.com/akka/akka/blob/master/akka-persistence-query/src/main/scala/akka/persistence/query/scaladsl/PersistenceIdsQuery.scala#L23)

### CODE REVIEW

File: docs/articles/persistence/persistent-fsm.md [hunk 1]
```diff
@@ -6,36 +6,73 @@ title: Persistence FSM
 
 `PersistentFSM` handles the incoming messages in an FSM like fashion. Its internal state is persisted as a sequence of changes, later referred to as domain events. Relationship between incoming messages, FSM's states and transitions, persistence of domain events is defined by a DSL.
 
-> [!WARNING]
-> PersistentFSM is marked as `experimental`.
```
Comment (alexvaluyskiy) [inline docs/articles/persistence/persistent-fsm.md]: Not experimental anymore

File: docs/api/index.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# PLACEHOLDER
+TODO: Add .NET projects to the *src* folder and run `docfx` to generate **REAL** *API Documentation*!
```
Comment (alexvaluyskiy) [inline docs/api/index.md]: A placeholder is better than 404 error

File: src/core/Akka.Persistence.Tests/Fsm/PersistentFSMSpec.cs [hunk 3]
```diff
@@ -414,6 +414,39 @@ public void PersistentFSM_must_allow_cancelling_stateTimeout_by_issuing_forMax_n
             probe.ExpectNoMsg(1.Seconds());
         }
 
+        [Fact]
+        public void PersistentFSM_must_save_periodical_snapshots_if_enablesnapshotafter()
+        {
+            var sys2 = ActorSystem.Create("PersistentFsmSpec2", ConfigurationFactory.ParseString(@"
+                akka.persistence.fsm.enable-snapshot-after = on
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/Fsm/PersistentFSMSpec.cs]: I copied this line from the JVM source, but we don't have this config

File: docs/api/index.md (hunk 2 above)
Comment (Aaronontheweb) [inline docs/api/index.md]: agree

File: docs/examples/DocsExamples/DocsExamples.csproj [hunk 4]
```diff
@@ -7,14 +7,19 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Akka" Version="1.2.0" />
-    <PackageReference Include="Akka.Cluster.Tools" Version="1.2.0" />
-    <PackageReference Include="Akka.Persistence" Version="1.2.0.36-beta" />
-    <PackageReference Include="Akka.TestKit.Xunit2" Version="1.2.0" />
+    <ProjectReference Include="..\..\..\src\core\Akka\Akka.csproj" />
```
Comment (Aaronontheweb) [inline docs/examples/DocsExamples/DocsExamples.csproj]: Good idea

File: src/core/Akka.Persistence.Tests/Fsm/PersistentFSMSpec.cs (hunk 3 above)
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.Tests/Fsm/PersistentFSMSpec.cs]: so is the feature itself not enabled? looks like you added it in the Akka.Persistence.cong

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy merged, nice work

**Pull request merged**